spec/unit/indirector/facts/facter_spec.rb

```diff
@@ -170,6 +170,12 @@
       @facter.find(@request)
     end
 
+    it 'should NOT add local facts' do
+      @facter.find(@request)
+
+      expect(facts).not_to have_received(:add_local_facts)
```

Comment:
[optional] maybe keep the same testing stile for consistency like the other examples:
 it 'description' do
	register expect
	execute action
end

eg: https://github.com/puppetlabs/puppet/pull/8399/files#diff-cb19d906cdb5e8716c1a61cdbfdf468b21e0709f02f6aef9a613470a4dcf8b83R168-R170

Reply:
Changed the test